core/src/program.rs

```diff
@@ -165,6 +165,11 @@ impl<EC: EvalCache> Program<EC> {
         })
     }
 
+    /// Only parse the program, don't typecheck or evaluate. returns the [`RichTerm`] AST
+    pub fn parse(&mut self) -> Result<RichTerm, Error> {
+        Ok(self.vm.import_resolver().parse_nocache(self.main_id)?.0)
```

Comment:
That is actually the context where I'm using it (parsing once), but I agree that the small bit of efficiency for that use-case is not worth the sacrifice to the other use-case. 

I was copying this from `pprint_ast()` for some reason. Is there a reason we don't use the cache there?

Reply:
Mmmh, probably the same reasoning as you just had :stuck_out_tongue: it might make sense to have a uniform behavior. Having a `Program::parse_nocache` is fine by me, what I found a bit sneaky here is that the method is called `parse`, thus you would probably assume it uses the cache by default.